src/components/timezone/DateAndTime.jsx

```
@@ -0,0 +1,458 @@
+const SCREEN_ID = "anaconda-screen-date-time";
+
+const DEFAULT_NTP_SERVERS = [
+    "ntp.fedoraproject.org", "pool.ntp.org", "time.cloudflare.com"
+];
+
+const DateAndTimeSection = ({ setSectionValid, timeFormat, timezone }) => {
+    const [autoDateTime, setAutoDateTime] = useState(true);
+    const [ntpSelectOpen, setNtpSelectOpen] = useState(false);
+    const [ntpServer, setNtpServer] = useState("");
+    const [ntpServers, setNtpServers] = useState([]);
+    const [datetime, setDatetime] = useState(DateTime.utc());
+
+    // Initial data fetch
+    useEffect(() => {
+        const updateTimeDateData = async () => {
+            try {
+                const ntpEnabled = await getNTPEnabled();
+                setAutoDateTime(!!ntpEnabled);
+
+                const dtIso = await getSystemDateTime();
+                if (dtIso) {
+                    setDatetime(
+                        DateTime.fromISO(dtIso, { zone: "utc" }).setZone(timezone)
+                    );
+                }
+
+                // DBus time sources (user-configured)
+                const srcs = await getTimeSources();
+                const ntps = srcs
+                        .filter(s => s.type?.v === "NTP")
+                        .map(s => s.name?.v)
+                        .filter(Boolean);
+
+                // Fallback: read from system config (chrony)
+                const srcConf = await getTimeServersFromConfig();
+                const confNtpServers = srcConf
+                        .map(s => s.hostname?.v)
+                        .filter(Boolean);
+
+                const finalServers = ntps.length
+                    ? ntps
+                    : (confNtpServers.length ? confNtpServers : DEFAULT_NTP_SERVERS);
+
+                setNtpServers(finalServers);
+                setNtpServer(finalServers[0] || "ntp.fedoraproject.org");
+            } catch (err) {
+                console.error("Failed to fetch time/date/time source data:", err);
```

We should not silently ignore errors here I think. If you dont handle it at all it would be cought by the global error handler if it fails, which I believe it better than ignoring.